action/action.go

```diff
@@ -41,17 +41,18 @@ type hasDestination interface {
 
 // Sign signs the action using sender's private key
 func Sign(act Envelope, sk crypto.PrivateKey) (SealedEnvelope, error) {
-	sealed := SealedEnvelope{Envelope: act}
-
-	sealed.srcPubkey = sk.PublicKey()
+	sealed := SealedEnvelope{
+		Envelope:  act,
+		srcPubkey: sk.PublicKey(),
+	}
 
 	hash := act.Hash()
 	sig, err := sk.Sign(hash[:])
 	if err != nil {
 		return sealed, errors.Wrapf(ErrAction, "failed to sign action hash = %x", hash)
 	}
 	sealed.signature = sig
-	sealed.payload.SetEnvelopeContext(sealed)
+	sealed.Action().SetEnvelopeContext(sealed)
```

Comment:
This is different from what I suggested in phone call. We may introduce a function "SetSealedContext" into Envelope, which is an interface after this PR. Then, we can call act.SetSealedContext(sealed).

Moreover, I am thinking about changing Envelope to be a SealedEnvelopeBuilder, say adding the following function to Envelope:
```Seal(sk crypto.PrivateKey) (*SealedEnvelope, error)```

Reply:
yes, i  did
see https://github.com/iotexproject/iotex-core/pull/2586/commits/080d457a696f83775c902cdfe09a69317c249261

if you want, can have another PR for adding SealedEnvelopeBuilder and Seal()

Reply:
updated: https://github.com/iotexproject/iotex-core/pull/2586/commits/631d6e9e8cad39d594cc2462d45aed4a7b632b11